libusb/hid.c

```diff
@@ -1582,8 +1666,31 @@ int HID_API_EXPORT hid_send_feature_report(hid_device *dev, const unsigned char
 		(unsigned char *)data, length,
 		1000/*timeout millis*/);
 
-	if (res < 0)
+	if (res < 0) {
+		wchar_t *custom_string = NULL;
+
+		switch (res) {
+		case LIBUSB_ERROR_TIMEOUT:
+			custom_string = L"Transfer timed out";
```

Comment:
why do you need a custom string here?
libusb has its own error string for that

Reply:
Because libusb specifies context-specific error messages in the documentation. "Device disconnected" instead of "broken pipe", for example. This context is only available at the call sites of each function, so only they can pass these contextual messages along.